src/App.tsx

```diff
@@ -12,13 +18,116 @@ const goodsFromServer: string[] = [
   'Honey',
   'Jam',
   'Garlic',
-];
-
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+].map(good => ({
+  name: good,
+  id: uuidv4(),
+}));
+
+type State = {
+  goods: Good[];
+  isListVisible: boolean;
+  isReversed: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isListVisible: false,
+    isReversed: false,
+  };
+
+  sortedByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.name.length - b.name.length),
+    }));
+  };
+
+  sortedByAlphbet = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.name.localeCompare(b.name)),
+    }));
+  };
+
+  showList = () => {
+    this.setState(state => ({
+      isListVisible: !state.isListVisible,
+    }));
+  };
```

Comment:
If function has name showList it's should only show but not hidde
```suggestion
  showList = () => {
    this.setState({ isListVisible: true })
  };
```